### ISSUE TITLE
Upgrade `maven-artifact-plugin` to a version that doesn't ban `flatten-maven-plugin`

### ISSUE THREAD

**Issue by pefernan** [apache/incubator-kie-tools#3981]
`kie-tools` (as all our kie environment) uses an old version of the `maven-artifact-plugin` (`3.4.1`) which could be upgrade to a newer version (at this time `3.5.1`).

`3.4.1` has the inconvenience that doesn't allow having *Reproducible Builds* if `maven-flatten-plugin` is present, and required a configuring the `maven-artifact-plugin` to exclude it, see https://github.com/apache/incubator-kie-tools/pull/2455.

As a part of this change we should remove the `<pluginIssues>` config and the `not-reproducible-plugins.properties` from `@kie-tools/maven-base`.

> IMPORTANT: at this time `maven-artifact-plugin` (`3.5.1`) requires `maven-quarkus-plugin` to be upgraded to `3.9.0` which is not LTS, so only do this upgrade after upgrading the `Quarkus` version too.

**Comment by gitgabrio** [comment]
@pefernan @tiagobento 
This is an extremely low priority, even with the 3.9.0 LTS, because
1. it involve change modification overall our codebase
2. it does not add - by itself - any value (unless we find out that is required by some other reasons)


Beside that, I would kindly ask anyone to stop creating "repo-specific" tickets for changes that should be cross-repo, as this one.

**Comment by pefernan** [comment]
@gitgabrio true. Just added it as a reminder to cleanup the reproducible build extra config I added in `kie-tools`

**Comment by tiagobento** [comment]
I think the value of this task is keeping our software up-to-date with the latest possible versions of the Maven toolchain. Keeping our own list of plugins that are known to not work for reproducible builds is a tech debt, and we might even get this upgrade for free when we move to Maven 4, for example. Who knows. Thanks for reporting it, great to have this mapped.
